packages/@tinacms/graphql/src/builder/index.ts

```diff
@@ -1176,17 +1175,24 @@ export class Builder {
     namespace: string[]
     collections: string[]
   }) => {
+    const fields = await sequential(
+      this.tinaSchema.getCollectionsByName(field.collections),
+      async (collection) => {
+        return astBuilder.InputValueDefinition({
+          name: collection.name,
+          type: NAMER.dataMutationTypeName([collection.name]),
+        })
+      }
+    )
+    fields.push(
```

Comment:
`_buildReferenceMutation` is used for both `createDocument` and `updateDocument` and looking at the resolve code below this would error if you provided it during the `createDocument`. Right now the mutations for create and update share the same GraphQL type but we should split these up into separate types and only have `relativePath` for the update mutation.

Reply:
Good catch. This has been fixed. It now has its own param type.